CHANGELOG.md

```diff
@@ -1,3 +1,10 @@
+# 2.0.0
+
+## What's Changed
+### Breaking Changes
+* Remove `MakeLogicSigAccount` and replace with `MakeLogicSigAccountEscrow`. Mark `MakeLogicSig` as a private function as well, only intended for internal use.
+* Rename `SignLogicsigTransaction` to `SignLogicSigTransaction`.
```

Comment:
@algochoi Can you add a CHANGELOG note to reflect the langspec + templates removal?

I offer a _suggestion_ based on the analogous py-algorand-sdk change.


```suggestion
* Rename `SignLogicsigTransaction` to `SignLogicSigTransaction`.
* Remove logicsig templates, `logic/langspec.json`, and all methods depending on it.
```